### PR TITLE
Fix multiple H1 tags on /docs/spec page

### PR DESCRIPTION
**Pull request by benwerner01**
This PR stops displaying multiple H1 tags on the specification pages.

It addresses this issue by re-structuring the specification sub-pages to stop universally displaying the specification banner, and only displaying it at the `/docs/spec` page.

All other sub-pages will display their parents in breadcrumbs, both on mobile and desktop.

Additional changes:
 - stop parsing the title of a page with sub-pages from a `title.txt` file, and instead look it up in the `0_index.mdx` file

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1202520215413533

### CODE REVIEW

**Comment by nonparibus** [comment]
This looks good to me visually. @nathggns @CiaranMn Could one of you code-review? Thanks.

File: site/src/components/page-sidebar.tsx [hunk 1]
```diff
@@ -462,31 +461,16 @@ export const Sidebar: VFC<SidebarProps> = ({
             opacity: 0,
           }}
         />
-        {pages.length > 1
```
Comment (nathggns) [inline site/src/components/page-sidebar.tsx:465]: Not sure I understand why this change is necessary – could you explain?

**Review by CiaranMn — COMMENTED** [review]
I've tested for functionality and all appears to be working correctly (highlighting correct section, jumping to anchors, etc). There are a few things in the diff I wasn't sure of the motivation for (changes not covered in the description), so let me know if there are things I should be specifically testing for I might've missed.

File: site/src/components/navbar/mobile-nav-items.tsx [hunk 2]
```diff
@@ -185,6 +170,21 @@ const MobileNavNestedPage = <T extends SiteMapPage | SiteMapPageSection>({
                 />
               ),
             )}
+            {itemIsPage(item) ? (
```
[the remark this answers is not in the document]
Comment (benwerner01) [reply site/src/components/navbar/mobile-nav-items.tsx:173]: Not sure the order was incorrect previously, but it was after the changes introduced by this PR.

The sub-sections of a page should be displayed before its sub-pages, so that for the "Specification" page the "Introduction" sub-section gets displayed before all the sub-pages:
image

File: site/src/components/page-sidebar.tsx (hunk 1 above)
Comment (benwerner01) [reply site/src/components/page-sidebar.tsx:465]: I'm not sure either actually, doesn't introduce any issues locally for me. I think I may have incorrectly thought it was causing an issue and then forget to revert the change, so I'm happy to revert it now.

Just want to clarify what this code was doing, as I'm still not entirely sure. We want pages that have a single sub-page to not display that sub-page directly, and instead render its sections directly? Or is there other reasoning for why we are special-casing this here?

Comment (nathggns) [reply site/src/components/page-sidebar.tsx:465]: I don't actually know either – I didn't write this. Looks like it was introduced in https://github.com/blockprotocol/blockprotocol/commit/3ea6f3b4b09b9a2491e068ef7afc63f7db3a2e5d – any ideas?

File: site/src/components/mdx/mdx-components.tsx [hunk 3]
```diff
@@ -82,9 +83,29 @@ export const mdxComponents: Record<string, FunctionComponent<any>> = {
...
+    >
+      {Array.isArray(children)
+        ? children.map((child) =>
+            child && typeof child === "object" && "props" in child
+              ? child.props.children
+              : child,
+          )
+        : children && typeof children === "object" && "props" in children
+        ? children.props?.children
+        : null}
```
[the remark this answers is not in the document]
Comment (nathggns) [reply site/src/components/mdx/mdx-components.tsx:106]: Ohhhh, that totally makes sense. I wonder if `next-mdx-remote` can be configured to stop this. Could you add a comment with more or less the above for now?

[the remark this answers is not in the document]
Comment (benwerner01) [reply site/src/components/mdx/mdx-components.tsx]: Have added a comment, and have thought of an alternative implementation which doesn't filter the react children and therefore seems less brittle: https://github.com/blockprotocol/blockprotocol/pull/410/commits/878d8fe7f19e5de394d6671ea73898f729f2bb2f

[the remark this answers is not in the document]
Comment (nathggns) [reply site/src/components/mdx/mdx-components.tsx]: Great!

**Review by nathggns — DISMISSED** [review]

**Review by nathggns — APPROVED** [review]
